metrics/influx.go

```diff
@@ -0,0 +1,165 @@
+package metrics
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/prebid/prebid-cache/config"
+	"github.com/rcrowley/go-metrics"
+	"github.com/vrischmann/go-metrics-influxdb"
+)
+
+type MetricsEntry struct {
+	Duration   metrics.Timer
+	Errors     metrics.Meter
+	BadRequest metrics.Meter
+	Request    metrics.Meter
+}
+
+type MetricsEntryByFormat struct {
+	Duration       metrics.Timer
+	Errors         metrics.Meter
+	BadRequest     metrics.Meter
+	JsonRequest    metrics.Meter
+	XmlRequest     metrics.Meter
+	DefinesTTL     metrics.Meter
+	InvalidRequest metrics.Meter
+	RequestLength  metrics.Histogram
+}
+
+type ConnectionMetrics struct {
+	ActiveConnections      metrics.Counter
+	ConnectionCloseErrors  metrics.Meter
+	ConnectionAcceptErrors metrics.Meter
+}
+
+func NewInfluxMetricsEntry(name string, r metrics.Registry) *MetricsEntry {
+	return &MetricsEntry{
+		Duration:   metrics.GetOrRegisterTimer(fmt.Sprintf("%s.request_duration", name), r),
+		Errors:     metrics.GetOrRegisterMeter(fmt.Sprintf("%s.error_count", name), r),
+		BadRequest: metrics.GetOrRegisterMeter(fmt.Sprintf("%s.bad_request_count", name), r),
+		Request:    metrics.GetOrRegisterMeter(fmt.Sprintf("%s.request_count", name), r),
+	}
+}
+
+func NewInfluxMetricsEntryBackendPuts(name string, r metrics.Registry) *MetricsEntryByFormat {
+	return &MetricsEntryByFormat{
+		Duration:       metrics.GetOrRegisterTimer(fmt.Sprintf("%s.request_duration", name), r),
+		Errors:         metrics.GetOrRegisterMeter(fmt.Sprintf("%s.error_count", name), r),
+		BadRequest:     metrics.GetOrRegisterMeter(fmt.Sprintf("%s.bad_request_count", name), r),
+		JsonRequest:    metrics.GetOrRegisterMeter(fmt.Sprintf("%s.json_request_count", name), r),
+		XmlRequest:     metrics.GetOrRegisterMeter(fmt.Sprintf("%s.xml_request_count", name), r),
+		DefinesTTL:     metrics.GetOrRegisterMeter(fmt.Sprintf("%s.defines_ttl", name), r),
+		InvalidRequest: metrics.GetOrRegisterMeter(fmt.Sprintf("%s.unknown_request_count", name), r),
+		RequestLength:  metrics.GetOrRegisterHistogram(name+".request_size_bytes", r, metrics.NewExpDecaySample(1028, 0.015)),
+	}
+}
+
+func NewInfluxConnectionMetrics(r metrics.Registry) *ConnectionMetrics {
+	return &ConnectionMetrics{
+		ActiveConnections:      metrics.GetOrRegisterCounter("connections.active_incoming", r),
+		ConnectionAcceptErrors: metrics.GetOrRegisterMeter("connections.accept_errors", r),
+		ConnectionCloseErrors:  metrics.GetOrRegisterMeter("connections.close_errors", r),
+	}
+}
+
+type InfluxMetrics struct {
+	Registry        metrics.Registry
+	Puts            *MetricsEntry
```

Comment:
I still like separating incoming puts, incoming gets, db puts, and db gets into their own metrics for Prometheus.

Reply:
Ok, rolling back